modules/secrets-store-azure.adoc

```diff
@@ -0,0 +1,278 @@
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: csi-secrets-store-provider-azure-cluster-role
+subjects:
+- kind: ServiceAccount
+  name: csi-secrets-store-provider-azure
+  namespace: openshift-cluster-csi-drivers
+---
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  namespace: openshift-cluster-csi-drivers
+  name: csi-secrets-store-provider-azure
+  labels:
+    app: csi-secrets-store-provider-azure
+spec:
+  updateStrategy:
+    type: RollingUpdate
+  selector:
+    matchLabels:
+      app: csi-secrets-store-provider-azure
+  template:
+    metadata:
+      labels:
+        app: csi-secrets-store-provider-azure
+    spec:
+      serviceAccountName: csi-secrets-store-provider-azure
+      hostNetwork: true
+      containers:
+        - name: provider-azure-installer
+          image: mcr.microsoft.com/oss/azure/secrets-store/provider-azure:v1.4.1
+          imagePullPolicy: IfNotPresent
+          args:
+            - --endpoint=unix:///provider/azure.sock
+            - --construct-pem-chain=true
+            - --healthz-port=8989
+            - --healthz-path=/healthz
+            - --healthz-timeout=5s
+          livenessProbe:
+            httpGet:
+              path: /healthz
+              port: 8989
+            failureThreshold: 3
+            initialDelaySeconds: 5
+            timeoutSeconds: 10
+            periodSeconds: 30
+          resources:
+            requests:
+              cpu: 50m
+              memory: 100Mi
+            limits:
+              cpu: 50m
+              memory: 100Mi
+          securityContext:
+            privileged: true
+          volumeMounts:
+            - mountPath: "/provider"
+              name: providervol
+      affinity:
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+            - matchExpressions:
+              - key: type
+                operator: NotIn
+                values:
+                - virtual-kubelet
+      volumes:
+        - name: providervol
+          hostPath:
+            path: "/var/run/secrets-store-csi-providers"
+      tolerations:
+      - operator: Exists
+      nodeSelector:
+        kubernetes.io/os: linux
+----
++
+[NOTE]
+====
+This configuration is modified from the configuration provided in the link:https://azure.github.io/secrets-store-csi-driver-provider-azure/docs/getting-started/installation/[Azure documentation] so that it works properly with {product-title}.
+====
++
+////
+TODO: Modified the above YAML with the following:
+
+- Added the ClusterRole and ClusterRoleBinding
+- Added namespace to the serviceaccount and daemonset
+- Changed securityContext to privileged on the container
+
+See if all of these are correct or not. Removed the "oc adm policy add-scc-to-use" step
+////
+
+.. Create the provider resources by running the following command:
++
+[source,terminal]
+----
+$ oc apply -f azure-provider.yaml
```

Comment:
oc apply -f https://raw.githubusercontent.com/Azure/secrets-store-csi-driver-provider-azure/master/deployment/provider-azure-installer.yaml -n openshift-cluster-csi-drivers

Reply:
@ropatil010 We decided not to rely on the YAML provided by azure/AWS and instead show the configuration directly in the docs here.

See my [comment above](https://github.com/openshift/openshift-docs/pull/62385/files#r1322064238) where I took what Azure provided and tried to modify it to have everything needed to work in OCP.

I also tried to align it with the procedure for AWS. So for example, I updated the YAML to set it to privileged, instead of running that `oc adm policy` command.

So what I wanted to see if you could do, is try testing it out with this method instead, to see if I have everything right, or if we need to make any adjustments here. For example - the question on whether the cluster role and cluster role binding resources are necessary.

Reply:
@ropatil010 I see that @xingxingxia explained below the outcome of this command, and that what I had was not enough. So I will go ahead and add a step for this. 

However, the commands for Azure and AWS are slightly different. I'd like to use the same format. 

**AWS**

```
$ oc adm policy add-scc-to-user privileged -z csi-secrets-store-provider-aws -n openshift-cluster-csi-drivers
```


**Azure**

```
$ oc adm policy add-scc-to-user privileged system:serviceaccount:openshift-cluster-csi-drivers:csi-secrets-store-provider-azure
```

I went with the AWS one for now and modified Azure's to be: `$ oc adm policy add-scc-to-user privileged -z csi-secrets-store-provider-azure -n openshift-cluster-csi-drivers`. Let me know if that works or if the other method is better for both.